md/api.py

```diff
@@ -0,0 +1,138 @@
+from django.db import connections
+from django.db.models import Count, Q
+
+from rest_framework import viewsets
+from rest_framework.decorators import detail_route
+from rest_framework.response import Response
+from rest_framework_extensions.cache.decorators import cache_response
+from rest_framework_extensions.key_constructor import bits
+from rest_framework_extensions.key_constructor.constructors import DefaultObjectKeyConstructor
+
+from md.models import Agency, Stop
+from md import serializers
+from md.utils import GroupedData
+
+
+GROUPS = {'A': 'ASIAN',
+          'B': 'BLACK',
+          'I': 'NATIVE AMERIOCAN',
+          'U': 'OTHER',
+          'W': 'WHITE',
+          'H': 'HISPANIC'}
```

Comment:
This is almost the same as models.ETHNICITY_CHOICES.  You might be able to use ``GROUPS = {k: v.upper() for k, v in ETHNICITY_CHOICES}`` to avoid duplicating code.

One of the differences is a typo in "AMERICAN", which would be fixed by using the recipe above.  The other difference is the discrepancy between "Unknown" in ETHNICITY_CHOICES and "OTHER" in GROUPS.  I think that is a bug in ETHNICITY_CHOICES; please change the string for 'U' in ETHNICITY_CHOICES to 'Other'.